src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/inventory/inventory-devices/inventory-devices.component.ts

```diff
@@ -158,24 +175,50 @@ export class InventoryDevicesComponent implements OnInit, OnChanges {
 
     this.filterInDevices = [...this.devices];
     this.updateFilterOptions(this.devices);
+    if (this.fetchInventory.observers.length > 0) {
+      this.fetchInventorySub = this.table.fetchData.subscribe(() => {
+        this.fetchInventory.emit();
+      });
+    }
+  }
+
+  ngOnDestroy() {
+    if (this.fetchInventorySub) {
+      this.fetchInventorySub.unsubscribe();
+    }
   }
 
   ngOnChanges() {
     this.updateFilterOptions(this.devices);
     this.filterInDevices = [...this.devices];
-    // TODO: apply filter, columns changes, filter changes
+    this.doFilter();
+  }
+
+  refresh() {
+    if (this.fetchInventory.observers.length > 0) {
+      this.fetchInventory.emit();
+    }
```

Comment:
Could this be removed? Could no find where it is being used.

Reply:
Yeah. Thanks!
I think I better reworking this PR after https://github.com/ceph/ceph/pull/32568, since some logic are moved.

Reply:
Fixed